src/NUnitFramework/framework/Internal/AsyncToSyncAdapter.cs

```diff
@@ -41,10 +47,10 @@ public static bool IsAsyncOperation(Delegate @delegate)
                 return (TResult?)awaiter.GetResult();
             }
         }
-
-        private static IDisposable? InitializeExecutionEnvironment()
+        private static IDisposable? InitializeExecutionEnvironment(TestExecutionContext? executionContext)
```

Comment:
Could this parameter be required? If not provided, the danger is that we may forget to pass it. Passing `context: null` explicitly in code helps raise awareness that we're omitting context.

Reply:
I didn't want to many changes in existing code. I have looked at all usages and the remaining didn't seem to need it.
But I agree that the suggested code makes it clearer then using the overload.
When I'm back near a PC next week, I will remove the overload and change all calls.

Reply:
Changes made.